libs/libarchopenfpga/src/bitstream_setting.cpp

```diff
@@ -106,6 +106,10 @@ std::vector<NonFabricBitstreamSetting> BitstreamSetting::non_fabric() const {
   return non_fabric_;
 }
 
+std::vector<PathBitSetting> BitstreamSetting::path_bit_settings() const {
```

Comment:
Considering the size of bitstreams (which can be easily >1M), returning a large vector is not efficient. 
Also, we should avoid to expose raw data through any API. Otherwise, there is no meaning to create a data structure for encapsulation purpose.

My suggestion is to create unique ids. For example: 

https://github.com/lnis-uofu/OpenFPGA/blob/aed082817ed3421ac0cfc45e305ba9191bf457ad/libs/libfpgabitstream/src/bitstream_manager.h#L116-L129

The API could be

```
std::string overwrite_bit_path(const OverwriteBitId& ow_bit_id) const;
std::string overwrite_bit_path(const OverwriteBitId& ow_bit_id) const;
```

Also, you need validator like: 

https://github.com/lnis-uofu/OpenFPGA/blob/aed082817ed3421ac0cfc45e305ba9191bf457ad/libs/libfpgabitstream/src/bitstream_manager.h#L217

Reply:
Now there is OverwriteBitstreamId implemented same as other, done via https://github.com/chungshien/OpenFPGA/commit/2a3d69aded840c54212dd9652670ab54e9ab1222